osidb/models/affect.py

```diff
@@ -483,6 +493,20 @@ def _validate_purl(self, **kwargs):
         except Exception as exc:
             raise ValidationError(f"Invalid purl '{self.purl}': {exc}")
 
+    def _validate_purl_and_ps_component(self, **kwargs):
+        """
+        Checks that ps_component matches the one included in purl if purl exists.
+        """
+        if (
+            self.purl
+            and self.ps_component
+            and self.ps_component != PackageURL.from_string(self.purl).name
+        ):
+            raise ValidationError(
+                "ps_component does not match the one included in purl: "
+                f"ps_component: {self.ps_component}, purl: {self.purl}."
```

Comment:
There should be the full affect identification (also PS module) so the user easily sees which one is incorrect.